pyproject.toml

```diff
@@ -53,6 +53,45 @@ boto3 = [
 Repository = "https://github.com/aio-libs/aiobotocore"
 Documentation = "https://aiobotocore.aio-libs.org"
 
+[dependency-groups]
+botocore-dev = [
+    # keep in sync with https://github.com/boto/botocore/blob/develop/requirements-dev.txt
+    # "wheel==0.43.0",
+    # "behave==1.2.5",
+    # "jsonschema==4.21.1",
+    "coverage==7.2.7",
+    # "setuptools==71.1.0;python_version>='3.12'",
+    # "packaging==24.1;python_version>='3.12'", # Requirement for setuptools>=71
+
+    # Pytest specific deps
+    "pytest==8.1.1",
+    "pytest-cov==5.0.0",
+    "pytest-xdist==3.5.0",
+    # "atomicwrites>=1.0", # Windows requirement
+    # "colorama>0.3.0", # Windows requirement
```

Comment:
what are these commented out packages for?

Reply:
As far as I can tell, those are dev dependencies from upstream that we do not currently need.

This PR does not change the dev dependencies in any way, it merely replaces the existing requirements-dev.in file into the [PEP 735](https://peps.python.org/pep-0735/) compliant [dependency-groups] table in pyproject.toml.

Reply:
I'm wondering if there's merit in having a dependency group for the upstream deps. You could just keep a requirements file in sync with the other repo if it's truly needed and then, extract whatever is relevant into this group, I suppose...

Reply:
We already have #1251 to track improvement of handling of upstream dev dependencies.

The motivation of following relevant upstream dev dependencies closely is that we attempt to keep upstream code in our codebase as close to verbatim as possible to simplify maintenance. Any drift in formatting, linter suppression comments and so forth is undesired.

I am proposing to split the existing `requirements-dev.in` file into two separate dependency groups to make the distinction between upstream dev dependencies and aiobotocore dev dependencies explicit for the first time. Just to avoid any confusion: These are not "extras" that leak into the published distribution packages in any way, it is merely a logical grouping of dev dependencies, intended to simplify their handling.